src/main.cc

```diff
@@ -203,18 +208,533 @@ void usage(int code = 1)
+
+	int signature_re_level = 4;
+	bool ignore_checksums = false;
+	bool use_watchdog = false;
+	double pseudo_realtime = 0;
+	DNS_MgrMode dns_mode = DNS_DEFAULT;
+
+	bool supervisor_mode = false;
+	bool parse_only = false;
+	bool bare_mode = false;
+	bool debug_scripts = false;
+	bool perftools_check_leaks = false;
+	bool perftools_profile = false;
+
+	bool run_unit_tests = false;
+	std::vector<std::string> doctest_args;
+
+	std::optional<std::string> pcap_filter;
+	std::vector<std::string> interfaces;
+	std::vector<std::string> pcap_files;
+	std::vector<std::string> signature_files;
+
+	std::optional<std::string> pcap_output_file;
+	std::optional<std::string> random_seed_input_file;
+	std::optional<std::string> random_seed_output_file;
+	std::optional<std::string> process_status_file;
+	std::optional<std::string> zeekygen_config_file;
+	std::string libidmef_dtd_file = "idmef-message.dtd";
+
+	std::set<std::string> plugins_to_load;
+	std::vector<std::string> scripts_to_load;
+	std::vector<std::string> script_options_to_set;
+
+	/**
+	 * Unset options that aren't meant to be used by the supervisor, but may
+	 * make sense for supervised nodes to inherit (as opposed to flagging
+	 * as an error an exiting outright if used in supervisor-mode).
+	 */
+	void filter_supervisor_options()
+		{
+		pcap_filter = {};
+		interfaces = {};
+		pcap_files = {};
+		signature_files = {};
+		pcap_output_file = {};
+		}
+
+	/**
+	 * Inherit certain options set in the original supervisor parent process
+	 * and discard the rest.
+	 */
+	void filter_supervised_node_options()
+		{
+		auto og = *this;
+		*this = {};
+
+		debug_log_streams = og.debug_log_streams;
+		debug_script_tracing_file = og.debug_script_tracing_file;
+		script_code_to_exec = og.script_code_to_exec;
+		script_prefixes = og.script_prefixes;
+
+		signature_re_level = og.signature_re_level;
+		ignore_checksums = og.ignore_checksums;
+		use_watchdog = og.use_watchdog;
+		pseudo_realtime = og.pseudo_realtime;
+		dns_mode = og.dns_mode;
+
+		bare_mode = og.bare_mode;
+		perftools_check_leaks = og.perftools_check_leaks;
+		perftools_profile = og.perftools_profile;
+
+		pcap_filter = og.pcap_filter;
+		signature_files = og.signature_files;
+
+		// TODO: These are likely to be handled in a node-specific or
+		// use-case-specific way.  e.g. interfaces is already handled for the
+		// "cluster" use-case, but don't have supervised-pcap-reading
+		// functionality yet.
+		/* interfaces = og.interfaces; */
+		/* pcap_files = og.pcap_files; */
+
+		pcap_output_file = og.pcap_output_file;
+		random_seed_input_file = og.random_seed_input_file;
+		random_seed_output_file = og.random_seed_output_file;
+		process_status_file = og.process_status_file;
+
+		plugins_to_load = og.plugins_to_load;
+		scripts_to_load = og.scripts_to_load;
+		script_options_to_set = og.script_options_to_set;
+		}
+};
+
+static void init_supervised_node(zeek_options* options)
```

Comment:
Move this out of `main.cc` into the supervisor code?

Reply:
Sure (mostly only there due to `zeek_options` dependency, but since that's also getting factored...)